hal/network/lwip/wiznet/wiznetif.cpp

```diff
@@ -50,7 +50,11 @@ LOG_SOURCE_CATEGORY("net.en")
 #endif /* WIZNET_SPI_MODE */
 
 #ifndef WIZNET_SPI_CLOCKDIV
+#if PLATFORM_ID == PLATFORM_BSOM || PLATFORM_ID == PLATFORM_ASOM || PLATFORM_ID == PLATFORM_XSOM
+#define WIZNET_SPI_CLOCKDIV 4
```

Comment:
Let's introduce a feature flag in `hal_platform.h` for this and set it for SoM platforms. Something along the lines of `HAL_PLATFORM_WIZNET_SPI_CLOCKDIV`.

Reply:
How about `HAL_PLATFORM_HW_FORM_FACTOR_SOM` instead?

Reply:
Nope, but if needed we can introduce if it makes certain things easier. This particular settings IMO should be an explicit value coming from `hal_platform.h` though.

Reply:
Should WIZNET really be in the name `HAL_PLATFORM_WIZNET_SPI_CLOCKDIV` or would it be better served to be generalized as `HAL_PLATFORM_SPI_CLOCKDIV`.

Then you could use `HAL_PLATFORM_SPI_CLOCKDIV` and `HAL_PLATFORM_WIZNET_SPI` in conjunction.

Reply:
Since this is a setting that specifically applies when we talk to the Wiznet chip, I'd say it should be `HAL_PLATFORM_WIZNET_SPI_CLOCKDIV`. We are not limiting the SPI clock speed in general, we are limiting it just for particular platforms for interaction with the Ethernet featherwing.

Reply:
I just keep getting the feeling that we are not abstracting around the right thing if we are calling out WIZNET specifically in _OUR_ `hal_platform.h`.

Wouldn't `HAL_PLATFORM_ETHERNET_FEATHERWING_SPI_CLOCKDIV` make better sense?

Then we could simplify the code in `hal/network/lwip/wiznet/wiznetif.cpp` to become...

```c++
#define WIZNET_SPI_CLOCKDIV HAL_PLATFORM_ETHERNET_FEATHERWING_SPI_CLOCKDIV
```

Reply:
Sure, it can be more generic. At the same time, there is no reason to be more specific. You can find specific defines that talk about e.g. BQ24195 PMIC.

Reply:
In that case, I would like to implement it as described in the comment above.

Reply:
No strong preferences. My choices was merely due to the class/source file being named wiznetxxx, so it kind of makes sense to base the setting on that as well so that it's less confusing.